Add documentations for Asgardeo Trigger. (#113)
ballerina-platform/asyncapi-triggers: asyncapi/asgardeo/Module.md

## Overview

> The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your [Asgardeo](https://wso2.com/asgardeo/) Organization. This functionality is provided by Asgardeo Events API.

Reviewer: ```suggestion
The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your [Asgardeo](https://wso2.com/asgardeo/) organization. This functionality is provided by Asgardeo Events API.
```

Author: Addressed via https://github.com/ballerina-platform/asyncapi-triggers/pull/113/commits/2ad7204cdcd395c42e6c4985b8b686f2296ed0e2

> The Asgardeo Trigger module allows you to listen to following events occur in your Asgardeo organization.

Reviewer: ```suggestion
The Asgardeo Trigger module allows you to listen to the following events when they occur in your Asgardeo organization:
```

Author: Addressed via https://github.com/ballerina-platform/asyncapi-triggers/commit/2ad7204cdcd395c42e6c4985b8b686f2296ed0e2

- Registration Events
   - Add User Event
   - Accept User Invite Event
   - Confirm Self Sign Up Event
- User Operation Events
   - User Account Lock Event
   - User Account Unlock Event
   - User Credential Update Event
   - User Delete Event
   - User Group Update Event
- Login Events
   - Login Success Event

## Prerequisites
- An organization in Asgardeo. Learn how to sign up and create an organization from [here](https://wso2.com/asgardeo/docs/get-started/create-asgardeo-account/#sign-up).
- An organization in Choreo created with the same email address and under the same name that was used to create the organization in Asgardeo.
- Enable event publishing in your Asgardeo Organization.

## Quickstart

### Step 1: Create a webhook in Choreo to receive Asgardeo events

> - To use the Asgardeo listener, please go the [Choreo Console](https://console.choreo.dev) and create a webhook.

Reviewer: ```suggestion
To use the Asgardeo listener, go to the [Choreo Console](https://console.choreo.dev) and create a webhook.
```

## Overview

> The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your [Asgardeo](https://wso2.com/asgardeo/) organization. This functionality is provided by the [Asgardeo Events API.](https://wso2.com/asgardeo/docs/references/asgardeo-events/)

Reviewer: ```suggestion
The `ballerinax/trigger.asgardeo` module provides a Listener to grasp events triggered from your [Asgardeo](https://wso2.com/asgardeo/) organization. This functionality is provided by the [Asgardeo Events API](https://wso2.com/asgardeo/docs/references/asgardeo-events/).
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

The Asgardeo Trigger module allows you to listen to the following events when they occur in your Asgardeo organization:

> - Registration Events

Reviewer: ```suggestion
- **Registration Events**
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

> - Registration Events

Reviewer: ```suggestion
- **Registration Events**
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

   - Add User Event
   - Accept User Invite Event
   - Confirm Self Sign Up Event

> - User Operation Events

Reviewer: ```suggestion
- **User Operation Events**
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

   - User Account Lock Event
   - User Account Unlock Event
   - User Credential Update Event
   - User Delete Event
   - User Group Update Event

> - Login Events

Reviewer: ```suggestion
- **Login Events**
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

   - Login Success Event

> Learn more about Asgardeo events from [here.](https://wso2.com/asgardeo/docs/references/asgardeo-events/#configure-asgardeo-to-publish-events)

Reviewer: ```suggestion
Learn more about [Asgardeo events](https://wso2.com/asgardeo/docs/references/asgardeo-events/#configure-asgardeo-to-publish-events).
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

## Prerequisites
- An organization in Asgardeo. Learn how to sign up and create an organization from [here](https://wso2.com/asgardeo/docs/get-started/create-asgardeo-account/#sign-up).
- An organization in [Choreo](https://wso2.com/choreo/) created with the same email address and under the same name that was used to create the organization in Asgardeo.
- Enable event publishing in your Asgardeo Organization. Learn how to enable from [here.](https://wso2.com/asgardeo/docs/references/asgardeo-events/#configure-asgardeo-to-publish-events)

## Quickstart

> 

Reviewer: ```suggestion
Follow the steps given below to try out the Asgardeo listener.
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

> 

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

### Step 1: Create a webhook in Choreo to receive Asgardeo events

To use the Asgardeo listener, go to the [Choreo Console](https://console.choreo.dev) and create a webhook  with **Asgardeo** as the trigger type.

> Learn more about how to develop a webhook from [here.](https://wso2.com/choreo/docs/webhook/#develop-a-webhook)

Reviewer: ```suggestion
Learn more about [how to develop a webhook](https://wso2.com/choreo/docs/webhook/#develop-a-webhook).
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

### Step 2: Program a logic to trigger for Asgardeo events

> - Now you can implement listener remote functions supported by this connector.

Reviewer: ```suggestion
Now you can implement listener remote functions supported by this connector.
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/pull/113/commits/fcd92a802f674aec9896fac8bc36af872d2f40c5

> - Write a remote function to receive a particular event type. 

Reviewer: ```suggestion
1. Write a remote function to receive a particular event type. 
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

> - Implement your logic within that function as shown in the below sample. This is an example of logging Asgardeo events in the Choreo console.

Reviewer: ```suggestion
2. Implement your logic within that function as shown in the below sample. This is an example of logging Asgardeo events in the Choreo console.
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5

    ```ballerina
    import ballerinax/trigger.asgardeo;
    import ballerina/log;
    import ballerina/http;
    
    configurable asgardeo:ListenerConfig config = ?;
    
    listener http:Listener httpListener = new(8090);
    listener asgardeo:Listener webhookListener =  new(config,httpListener);
    
    service asgardeo:RegistrationService on webhookListener {
      
        remote function onAddUser(asgardeo:AddUserEvent event ) returns error? {
            log:printInfo(event.toJsonString());
        }
      
        remote function onConfirmSelfSignup(asgardeo:GenericEvent event ) returns error? {
            log:printInfo(event.toJsonString());
        }
      
        remote function onAcceptUserInvite(asgardeo:GenericEvent event ) returns error? {
            log:printInfo(event.toJsonString());
        }
    }
    
    service /ignore on httpListener {}
    ```

### Step 3: Deploy the Webhook in Choreo

> - Then navigate to the deploy tab and click on the Config & Deploy button to start the deploy process. Learn more about how to deploy a webhook from [here.](https://wso2.com/choreo/docs/webhook/#deploy-a-webhook)

Reviewer: ```suggestion
Once you have created the webhook, go to the **Deploy** tab on the Choreo Console and click on **Config & Deploy** to start the deployment process. Learn more about how to deploy a webhook from [here](https://wso2.com/choreo/docs/webhook/#deploy-a-webhook).
```

Author: Addressed with https://github.com/ballerina-platform/asyncapi-triggers/commit/fcd92a802f674aec9896fac8bc36af872d2f40c5
